features/22-email-verification/implementation.md

```diff
@@ -0,0 +1,216 @@
+# Implementation for verify by email and SMS
+
+## Server
+
+The server (core) will implement the following set of APIs to support
+verify by email and verify by SMS
+
+### `auth:verify:check`
+
+#### Overview
+
+#### Request
+
+* `verify_key` (string)
+  The auth key, either `email` or `phone`.
+
+* `user_id` (string, optional)
+  The user to verify. If the request is authenticated with access token
+  only the current user is accepted. If request is authenticated with master key
+  any user is accepted. Default to current user. If the request is not
+  authenticated, the user will be looked up from `verify_data`.
+
+* `verify_data` (string, optional)
```

Comment:
Would like to know what will happen if `verify_data` is different from the current user email or phone, those will be udpated?

Reply:
the check will not be successful